pkg/client/client.go

```diff
@@ -221,6 +235,45 @@ func (c *Client) runBuild(ctx context.Context, r interface{}, path, plandir, sdk
 	return c.request(ctx, "POST", path, rd, "Content-Type", contentType)
 }
 
+// getFilteredDirectory filters the directory dir according to the
+// ignored files specified in $dir/.testgroundignore. Returns a new
+// directory and a boolean indicating if such directory is temporary
+// and should be cleaned by the caller.
+func getFilteredDirectory(dir string) (string, bool, error) {
+	ignoreFilePath := filepath.Join(dir, ".testgroundignore")
+	_, err := os.Stat(ignoreFilePath)
+
+	if os.IsNotExist(err) {
+		return dir, false, nil
+	} else if err != nil {
+		return "", false, err
+	}
+
+	tgIgnore, err := ignore.CompileIgnoreFile(ignoreFilePath)
```

Comment:
Could we have a comment that explains that the output of `CompileIgnoreFile` is a set of regex patterns (as per https://github.com/sabhiram/go-gitignore/blob/master/ignore.go#L164)?

It wasn't immediately obvious to me and without this information it's hard to judge if the comparisons we're doing later are correct.

Reply:
I agree with @galargh  - I would like to see a more comments detailing this function.

Reply:
Updated

Reply:
Added a comment with some clarification.